agent/agent.go

```diff
@@ -3839,3 +3852,14 @@ func defaultIfEmpty(val, defaultVal string) string {
 	}
 	return defaultVal
 }
+
+// getUIConfig is the canonical way to read the value of the UIConfig at
+// runtime. It is thread safe and returns the most recent configuration which
+// may have changed since the agent started due to config reload.
+func (a *Agent) getUIConfig() config.UIConfig {
```

Comment:
I think we should be avoiding new Agent methods as much as possible. How will this be used? Could this be a method on `HTTPServer` instead? I assume it will be used there?

Reply:
#8623 has a pattern for config reloading that I think might work better. It would remove the need for the extra field and method from `Agent`. It introduces an interface so that each component can reload its config.

Reply:
Yeah I didn't love putting this here but it seemed cleaner than attaching the atomic UI config state to some other component - I considered putting it on the HTTPServer itself but that felt wrong as the HTTPServer is really just a thin wrapper around the HTTP handler functions and not something that semantically owns the config for the UI.

I like your inverted reload patten in #8623 - I think I would use that once it's landed, however I still feel odd about leaving this config state with no canonically correct way to read it unless you happened to only need it on a reload. Especially since it's already confusing that there are two different places this config can be read from. It seems like it would be more likely to be mis-used (either read from the runtime config instead of the reloaded one or duplicated atomic logic) in the future if there was no canonical accessor on the object that owns the state.

I think overall my ideal approach here would be to make the whole RuntimeConfig atomically reloadable since the config package is really the most semantic place to put logic related to config and reloading, however that's a big enough task not to want to have on critical path for a feature we're trying to ship for now! It requires changing every part of code that accesses the agent config! But it would avoid the confusion you identified below where only parts of the config actually change during a reload and code that refers only to the original config won't ever observe the reload even if it's looking at a "reloadable" value.

Reply:
Actually, I'm working on the next two components and realise this reloaded UI needs to get referenced in more places and passing a whole agent around gets even more gross than it should be.

I think the pattern you proposed is cleaner and I'll probably switch to that.